code/modules/fishing/fish/types/air_space.dm

```diff
@@ -149,6 +159,35 @@
 	RegisterSignal(src, COMSIG_FISH_FINISH_GROWING, PROC_REF(on_growth))
 	update_appearance(UPDATE_OVERLAYS)
 
+/obj/item/fish/baby_carp/suicide_act(mob/living/user)
+	user.visible_message(span_suicide("[user] swallows [src] whole!"))
+	src.forceMove(user)
+	if(status == FISH_DEAD)
+		user.emote("gasp")
+		user.visible_message(span_suicide("[user] chokes on [src] and dies!"))
+		return OXYLOSS
+
+	// the fish grows
+	addtimer(CALLBACK(src, PROC_REF(gestation), user), 20 SECONDS)
+	user.visible_message(span_suicide("[user] starts growing unnaturally..."))
+
+	var/matrix/M = matrix()
+	M.Scale(1.8, 1.2)
+	animate(user, time = 20 SECONDS, transform = M, easing = SINE_EASING)
+	return MANUAL_SUICIDE
+
+/obj/item/fish/baby_carp/proc/gestation(mob/living/user)
```

Comment:
since this is a whole 20 seconds after the fact, we should probably check our user hasnt been deleted during that wait